### ISSUE TITLE
vim freezes

### ISSUE THREAD

**Issue by Sciss** [ensime/ensime-vim#164]
I am getting random freezes, even from just opening a file in vim.

    $ python --version
    Python 2.7.9
    $ vi -version
    VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Mar 31 2015 23:35:48)
    $ uname -v
    #1 SMP Debian 3.16.5-1 (2014-10-10)

sbt global plugins:

    addSbtPlugin("org.ensime" % "ensime-sbt" % "0.3.3")

I installed the following way:

    $ sudo apt-get install python-pip
    
    $ curl -fLo ~/.vim/autoload/plug.vim --create-dirs \
        https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim

    $ cat ~/.vimrc

```
set number          " Show line numbers.
set mouse=a         " Enable the use of the mouse.
filetype plugin indent on
filetype on
set autoindent " Copy indent from the row above
set si " Smart indent

set wildmenu

set nocompatible "Not vi compativle (Vim is king)

""""""""""""""""""""""""""""""""""
" Syntax and indent
"""""""""""""""""""""""""""""""""""
syntax on " Turn on syntax highligthing
set showmatch  "Show matching bracets when text indicator is over them

set hls " highlight search
set lbr " linebreak

" Use 2 space instead of tab during format
set expandtab
set shiftwidth=2
set softtabstop=2

" let g:solarized_termcolors=256
" set background=dark
" colorscheme solarized

call plug#begin('~/.vim/plugged')
Plug 'ensime/ensime-vim'
call plug#end()
```

Example:

    $ git clone https://github.com/Sciss/ScalaAudioFile.git
    $ cd ScalaAudioFile
    $ sbt gen-ensime
    $ vim src/main/scala/de/sciss/synth/io/package.scala

After a moment, status reads `Execute :EnClasspath to set a classpath`.

I try `:q` to quit. vim freezes and doesn't react to any keyboard input or ctrl-C, ctrl-D, can only be killed with `kill -9 <pid>`.

**Comment by jvican** [comment]
There is a current problem in which I have stumbled upon several times: when you open a vim instance you cannot quit when `ensime-vim` is waiting for server reply (which is the case in the beginning of the execution). You have to wait until the messages 'Ensime analyzer is ready' and 'Ensime indexer is ready'.
  
Why? No idea, we have to find it out.
  
I don't know if your problem is related to this bug but it probably is. We'll see how it ends up. I don't know when I'll have time to do this (any help is welcome) so I would recommend you to use neovim for some days if you are really interesting in using this plugin. `Ensime-vim` works way better on neovim than vim.

**Comment by Sciss** [comment]
Ok, it's not urgent. Actually I want to see a bit how the ensime clients work and if ensime is the right technology to use for an upgrade of my swing editor ;)

**Comment by jvican** [comment]
Nice, any contribution is appreciated @Sciss :smile:

**Comment by timothykim** [comment]
I have the exact same issue. Will be monitoring this issue.

**Comment by ChrisCoffey** [comment]
Likewise. I'm running NVIM 0.1.1 on OSX with python 2.7

**Comment by mantognini** [comment]
I wanted to try this plugin but I'm facing the same issue. The "fix" from #168 / @ches, i.e. `rm .ensime`, works but then I get no feedback from this plugin (or at least I think that's the reason why functions such as EnType display no result).

**Comment by jvican** [comment]
@mantognini You have to run `sbt gen-ensime` again. The `ensime-vim` plugin will only start working when the file `.ensime` is found (it's needed by the ensime server).

**Comment by mantognini** [comment]
@jvican Yes, I tried that but then I was back to the previous situation where it freezes.

**Comment by jvican** [comment]
OS? Python version?
  
Go to `$HOME/.config/ensime-vim/${your scala version}/` and execute `sbt saveClasspath`. Update completely your plugin from upstream. Try again.

**Comment by jvican** [comment]
Please, give me your vim version as well, just for the record.

**Comment by mantognini** [comment]
Ok, I'll do that either later today or tomorrow. Thanks for your help. :-)

**Comment by jvican** [comment]
You welcome. BTW, I'm also working at LAMP, so if you need me I'm at BC360.

**Comment by mantognini** [comment]
I'm not on campus today but here's the requested information:

 - OS: OS X 10.10.5
 - Python: 2.7.11
 - Vim: build from `brew` today, similar issues with gvim. [Detailed version information](https://gist.github.com/mantognini/d9d47420d089446a2f9b).
 - `~/.config/ensime-vim/` doesn't exist on my system and couldn't find a similar directory in my home folder:

```shell
~> find . -name 'ensime*' -type d
./.atom/.apm/ensime-client
./.atom/.apm/registry.npmjs.org/ensime-client
./.atom/packages/Ensime/ensime_update_coursier
./.atom/packages/Ensime/node_modules/ensime-client
./.coursier/cache/v1/https/repo1.maven.org/maven2/org/ensime
./.coursier/cache/v1/https/repo1.maven.org/maven2/org/ensime/ensime_2.11
./.ivy2/cache/scala_2.10/sbt_0.13/org.ensime/ensime-sbt
./.vim/bundle/ensime-vim
./.vim/bundle/ensime-vim/ensime_shared
```

Let me know if I can test anything else.

**Comment by jvican** [comment]
Ok, I fear that you'll need to wait until we fix that `vim` bug. In the meanwhile, if you need to get this working, I'd recommend you to use `neovim` which works perfectly with `ensime-vim`.

**Comment by mantognini** [comment]
I'll keep an eye on this issue then. (I've actually setup Atom to use ensime for the time being.)

**Comment by jvican** [comment]
There's been some good work on `ensime-vim` and there's a high probability this has already been fixed. Please try again, and remember to follow the troubleshooting guide in case of any problem. Beware that if there's no reply in the next days, I'm closing this!

**Comment by mantognini** [comment]
Maybe I did something wrong but I'm still facing the same issue; I can't quit vim without `kill -9 <pid>`. Also, I seem unable to launch the server -- `.ensime_cache/server.log` reads `Error: Could not find or load main class org.ensime.server.Server`. I tried the troubleshooting steps without luck. Note that I'm not using the latest vim. Unfortunately, I won't have time to investigate further soon, so if no one else reports similar results, feel free to close it as it might be something on my system.
